packages/jest-runtime/src/index.ts

```diff
@@ -1273,7 +1273,9 @@ class Runtime {
     moduleRequire.resolve = resolve;
     moduleRequire.cache = (() => {
       const notPermittedMethod = () => {
-        console.warn('`require.cache` modification is not permitted');
+        this._environment.global.console.warn(
```

Comment:
Not related, but noticed this when debugging.

before:
![image](https://user-images.githubusercontent.com/1404810/79916240-adf08200-8428-11ea-893b-2c7620aa6980.png)

after: 
![image](https://user-images.githubusercontent.com/1404810/79916260-b8128080-8428-11ea-961a-27abec4c9029.png)

Reply:
@SimenB, looks reasonable. We could move up `notPermittedMethod` helper to the `jest-resolve` root and wrap it with `once`, so we get just one warning until `jest-resolve` reload.